evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -191,3 +228,159 @@ class Pipeline(pipeline_class):
+
+
+@pytest.mark.parametrize("pipeline_class,objectives", [(TimeSeriesBinaryClassificationPipeline, ["MCC Binary"]),
+                                                       (TimeSeriesBinaryClassificationPipeline, ["Log Loss Binary"]),
+                                                       (TimeSeriesBinaryClassificationPipeline, ["MCC Binary", "Log Loss Binary"]),
+                                                       (TimeSeriesMulticlassClassificationPipeline, ["MCC Multiclass"]),
+                                                       (TimeSeriesMulticlassClassificationPipeline, ["Log Loss Multiclass"]),
+                                                       (TimeSeriesMulticlassClassificationPipeline, ["MCC Multiclass", "Log Loss Multiclass"]),
+                                                       (TimeSeriesRegressionPipeline, ['R2']),
+                                                       (TimeSeriesRegressionPipeline, ['R2', "Mean Absolute Percentage Error"])])
+@pytest.mark.parametrize("use_ww", [True, False])
+def test_score_works(pipeline_class, objectives, use_ww, X_y_binary, X_y_multi, X_y_regression):
+
+    preprocessing = ['Delayed Feature Transformer']
+    if pipeline_class == TimeSeriesRegressionPipeline:
+        components = preprocessing + ['Random Forest Regressor']
+    else:
+        components = preprocessing + ["Logistic Regression Classifier"]
+
+    class Pipeline(pipeline_class):
+        component_graph = components
+
+    pl = Pipeline({"pipeline": {"gap": 1, "max_delay": 2, "delay_features": False},
+                   components[-1]: {'n_jobs': 1}})
+    if pl.problem_type == ProblemTypes.TIME_SERIES_BINARY:
+        X, y = X_y_binary
+        y = pd.Series(y).map(lambda label: "good" if label == 1 else "bad")
+        expected_unique_values = {"good", "bad"}
+    elif pl.problem_type == ProblemTypes.TIME_SERIES_MULTICLASS:
+        X, y = X_y_multi
+        label_map = {0: "good", 1: "bad", 2: "best"}
+        y = pd.Series(y).map(lambda label: label_map[label])
+        expected_unique_values = {"good", "bad", "best"}
+    else:
+        X, y = X_y_regression
+        y = pd.Series(y)
+        expected_unique_values = None
+    if use_ww:
+        X = ww.DataTable(X)
+        y = ww.DataColumn(y)
+
+    pl.fit(X, y)
+    if expected_unique_values:
+        # NaNs are expected because of padding due to max_delay
+        assert set(pl.predict(X, y).dropna().unique()) == expected_unique_values
+    pl.score(X, y, objectives)
+
+
+@patch('evalml.pipelines.TimeSeriesClassificationPipeline._decode_targets')
+@patch('evalml.objectives.BinaryClassificationObjective.decision_function')
+@patch('evalml.pipelines.components.Estimator.predict_proba', return_value=pd.DataFrame({0: [1.]}))
+@patch('evalml.pipelines.components.Estimator.predict', return_value=pd.Series([1.]))
+def test_binary_classification_predictions_thresholded_properly(mock_predict, mock_predict_proba,
```

Comment:
This is to achieve 100% coverage on the binary classification `_predict` method.